File: my-folder/0013-roman-to-integer/solution.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans=0
        numArr=[]

        for _ in s:
            if _ == "I":
                numArr.append(1)
            if _ == "V":
                numArr.append(5)
            if _ == "X":
                numArr.append(10)
            if _ == "L":
                numArr.append(50)
            if _ == "C":
                numArr.append(100)
            if _ == "D":
                numArr.append(500)
            if _ == "M":
                numArr.append(1000)
        
         

        for i in range(len(numArr)):
            if i<len(numArr)-1:
                if (numArr[i]<numArr[i+1]):
                    ans-=int(numArr[i])
                else:
                    ans+=int(numArr[i])
            else:
                ans+=int(numArr[i])
        
        return ans

File: my-folder/0013-roman-to-integer/solution.cpp
class Solution {
public:
    int toNumber(char c)
    {
        switch(c)
        {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
        }
        return 0;
    }

    int romanToInt(string s) {
        
        int n = s.size();
        int res = 0;
        for (int i = 0; i < n; i++)
        {
            if (i < n - 1 && toNumber(s[i]) < toNumber(s[i + 1]))
            {
                res -= toNumber(s[i]);
            }
            else
            {
                res += toNumber(s[i]);
            }
        }
        return res;
    }
};
